### ISSUE TITLE
cached data is changed by other query

### ISSUE THREAD

**Issue by tsingsun** [ariga/entcache#20]
A developer told me there seemed to be a bug in EntCache, and I tried to reproduce it : 

https://github.com/tsingsun/entbug-entcache

i dont use docker compose file, and This issue is in mysql7 and 8 in my  case.

please run the testBugMySQL in `bug_test.go`

test code
```go
func test(t *testing.T, client *ent.Client) {
      .........
      for i := 0; i < 2; i++ {
		datas, err := client.SecurityPosition.Query().Where(securityposition.AccountIDIn(3)).
			Order(ent.Asc(securityposition.FieldProjectID, securityposition.FieldProductID)).
			All(context.Background())
		if err != nil {
			t.FailNow()
		}
                // first query the cache is correct, 
		if datas[0].ID == 8470 && *datas[0].MaterialNo != "688280.SH" {
			t.Error(i, *datas[0].MaterialNo)
		}
		if datas[1].ID == 8471 && *datas[1].MaterialNo != "300472.SZ" {
			t.Error(i, *datas[1].MaterialNo)
		}
                // When the following query is executed, the datas[0].MaterialNo in the cache is changed
		client.SecurityJournal.Query().
			Where(securityjournal.AccountIDIn(3),
				securityjournal.ChangeType("2"), securityjournal.IsDayBooking("Y"),
			).Order(ent.Desc(securityjournal.FieldID)).
			AllX(context.Background())
	}
```
test result
```output
=== RUN   TestBugMySQL
=== RUN   TestBugMySQL/8
    bug_test.go:95: 1 efx_rate
    bug_test.go:98: 1 tl_margin
--- FAIL: TestBugMySQL (0.03s)
    --- FAIL: TestBugMySQL/8 (0.03s)
```
I don't know if I have a usage problem, bug the problem product same time in other developer

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.18.3  |
| Ent         | 0.11.1   |
| Database    | MySQL   |
| Driver      | https://github.com/go-sql-driver/mysql   1.6.0     |

**Comment by tsingsun** [comment]
@a8m CC

**Comment by wanrui** [comment]
#16

**Comment by tsingsun** [comment]
right, mysql driver should be reused the driver.Value

**Comment by a8m** [comment]
Hey! I just merged #16 

Please, update to master and let me know if there are still issues with it. Thanks and sorry for the delay 🙏
